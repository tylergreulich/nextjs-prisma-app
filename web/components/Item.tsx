import * as React from 'react';
import Link from 'next/link';

import { ItemProps } from '../graphql/schemaTypes';
import { ItemStyles } from './styles/ItemStyles';
import { Title } from './styles/Title';
import { PriceTag } from './styles/PriceTag';
import { formatMoney } from '../lib/formatMoney';

export class Item extends React.Component<ItemProps, {}> {
  render() {
    const { id, title, price, description, image } = this.props.item;

    return (
      <ItemStyles>
        {image && <img src={image} alt={title} />}
        <Title>
          <Link href={{ pathname: '/item', query: { id } }}>
            <a>{title}</a>
          </Link>
        </Title>
        <PriceTag>{formatMoney(price)}</PriceTag>
        <p>{description}</p>
        <div className="buttonList" data-testId="buttonList">
          <Link href={{ pathname: 'update', query: { id } }}>
            <a>Edit</a>
          </Link>
          <button>Add to Cart</button>
          <button>Remove from Cart</button>
        </div>
      </ItemStyles>
    );
  }
}
