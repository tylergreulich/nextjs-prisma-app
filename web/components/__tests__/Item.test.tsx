import * as React from 'react';

import { render } from 'react-testing-library';

import { Item } from '../Item';

import { ItemProps } from '../../interfaces/Item.interface';

const fakeItem: ItemProps = {
  id: '1',
  title: 'A cool item',
  price: 5000,
  description: 'This item is really cool!',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg'
};

describe('<Item />', () => {
  it('renders and matches the snapshots', () => {
    const { getByAltText, getByText } = render(<Item {...fakeItem} />);
    const image = getByAltText(fakeItem.title) as HTMLImageElement;
    const priceTag = getByText('$50') as HTMLSpanElement;
    const title = getByText(fakeItem.title) as HTMLAnchorElement;

    expect(image).toMatchSnapshot();
    expect(priceTag).toMatchSnapshot();
    expect(title).toMatchSnapshot();
  });

  it('renders the buttons properly', () => {
    const { getByTestId } = render(<Item {...fakeItem} />);
    const buttonList = getByTestId('buttonList');
    expect(buttonList.children).toHaveLength(3);
  });
});
