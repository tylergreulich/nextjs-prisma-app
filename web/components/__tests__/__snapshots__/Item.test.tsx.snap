// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshots 1`] = `
<img
  src="dog.jpg"
  alt="A cool item"
/>
`;

exports[`<Item /> renders and matches the snapshots 2`] = `
<span
  class="sc-htpNat iCcKty"
>
  $50
</span>
`;

exports[`<Item /> renders and matches the snapshots 3`] = `
<a
  href="/item?id=abc123"
>
  A cool item
</a>
`;
